### ISSUE TITLE
Issues With Pseudo Parameters When Using LanguageExtensions

### ISSUE THREAD

**Issue by nVlast** [aws-cloudformation/cfn-lint#2811]
### CloudFormation Lint Version

0.79.1

### What operating system are you using?

Debian

### Describe the bug

`cfn-lint` raises various errors when using pseudo parameters along with the LanguageExtensions macro. I'll provide two examples:
- `AWS::Region` 
  ```yaml
  Transform: AWS::LanguageExtensions
  Resources:
    MyNestedStack:
      Type: AWS::CloudFormation::Stack
      Properties:
        TemplateURL: !Sub https://mybucket-${AWS::Region}.s3.${AWS::Region}.amazonaws.com/my-nested-stack.yml
  ```
  ```
  ❯ cfn-lint nested.yml 
  E1029 Found an embedded parameter "${AWS::Region}" outside of an "Fn::Sub" at Resources/MyNestedStack/Properties/TemplateURL
  nested.yml:6:7
  ```
- `AWS::AccountId`
  ```yaml
  Transform: AWS::LanguageExtensions
  Resources:
    MyBucket:
      Type: AWS::S3::Bucket
      Properties:
        BucketName: !Sub "my-bucket-${AWS::AccountId}"
  ```
  ```
  ❯ cfn-lint bucket.yml 
  E1029 Found an embedded parameter "${AWS::AccountId}" outside of an "Fn::Sub" at Resources/MyBucket/Properties/BucketName
  bucket.yml:6:7
  
  E3031 BucketName contains invalid characters (Pattern: ^[a-z0-9][a-z0-9//.//-]*[a-z0-9]$) at Resources/MyBucket/Properties/BucketName
  bucket.yml:6:7
  ```

For both of the above examples, remove `Transform: AWS::LanguageExtensions` removes the errors.

### Expected behavior

We use a template like the one with the nested stack in production and has been working flawlessly and even used to be deemed valid by previous cfn-lint versions so I would expect that newer versions of cfn-lint still think it's valid.

### Reproduction template

```yaml
Transform: AWS::LanguageExtensions
Resources:
  MyNestedStack:
    Type: AWS::CloudFormation::Stack
    Properties:
      TemplateURL: !Sub https://mybucket-${AWS::Region}.s3.${AWS::Region}.amazonaws.com/my-nested-stack.yml
```

**Comment by greenaussie** [comment]
I also noticed this just now.

**Comment by kddejong** [comment]
@nVlast this should be fixed.  Can you validate?

**Comment by andrewlangemann** [comment]
0.79.3 fixed the issue for me

**Comment by kddejong** [comment]
I believe this to be fixed we can re-open as needed.

**Comment by wmartins** [comment]
Hi @kddejong, thanks for working on this! Do you have any idea when this lands on homebrew? The current version there is `0.79.1`

**Comment by nVlast** [comment]
> @nVlast this should be fixed. Can you validate?

Thank you, @kddejong. The issue is now fixed!
